### ISSUE TITLE
Oro\Bundle\SyncBundle\WebSocket\Client\Rfc6455::connect() can block the application

### ISSUE THREAD

**Issue by dmitry-fedyuk** [oroinc/platform#700]
See the code:
https://github.com/orocrm/platform/blob/2.1.2/src/Oro/Bundle/SyncBundle/WebSocket/Client/Rfc6455.php#L50

```
$serverHandshake = fread($this->socket, 32);
```

I think, some timeout need be set before the `fread` call, e.g.:

```
stream_set_timeout($this->socket, 3);
```
Otherwise, `fread` can wait forever.

**Comment by d-beekeeper** [comment]
Hello dmitry-fedyuk.
Thanks for the notice! We will fix it during next release or two. Or, as an option, you can create your own pull request with fix.

**Comment by dmitry-fedyuk** [comment]
It is fixed in 2.5.0:
https://github.com/oroinc/platform/blob/2.5.0/src/Oro/Bundle/SyncBundle/WebSocket/Client/Rfc6455.php#L232
